compiler/pipes/deduce-implicit-types-and-casts.cpp

```diff
@@ -559,6 +562,84 @@ void DeduceImplicitTypesAndCastsPass::on_phpdoc_for_var(VertexAdaptor<op_phpdoc_
   phpdocs_for_vars.emplace_front(v_phpdoc);
 }
 
+void DeduceImplicitTypesAndCastsPass::patch_call_args(VertexAdaptor<op_func_call> &call, VertexRange f_called_params) {
+  auto call_args = call->args();
+
+  auto find_corresponding_param = [&f_called_params](const std::string &call_arg_name) -> std::optional<VertexAdaptor<op_func_param>> {
+  for (auto param: f_called_params) {
+    if (param.as<op_func_param>()->var()->get_string() == call_arg_name) {
+      return param.as<op_func_param>();
+    }
+  }
+  return std::nullopt;
+  };
+
+  std::vector<VertexAdaptor<op_func_param>> call_arg_to_func_param(call_args.size());
+  int call_arg_idx = 0;
+
+  // positional args
+  while (call_arg_idx < call_args.size() && call_arg_idx < f_called_params.size() && call_args[call_arg_idx]->type() != op_named_arg) {
+    call_arg_to_func_param[call_arg_idx] = f_called_params[call_arg_idx].as<op_func_param>();
+
+    if (call_arg_idx < f_called_params.size() && f_called_params[call_arg_idx]->extra_type == op_ex_param_variadic) {
+      int vararg_idx = call_arg_idx;
+      while (call_arg_idx < call_args.size()) {
+        call_arg_to_func_param[call_arg_idx] = f_called_params[vararg_idx].as<op_func_param>();
+        call_arg_idx++;
```

Comment:
Maybe it's better to use prefix addition here? Why do we need an extra copy?

```c++
++call_arg_idx;
```
Below also.